assets/src/edit-story/app/story/useStoryReducer/actions.js

```diff
@@ -0,0 +1,119 @@
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex } } );
+
+const deleteCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_PAGE, payload: { pageIndex: null } } );
+
+const updatePageProperties = ( dispatch ) => ( { pageIndex, properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex, properties } } );
+
+const updateCurrentPageProperties = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_PAGE, payload: { pageIndex: null, properties } } );
+
+const duplicatePage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex } } );
+
+const duplicateCurrentPage = ( dispatch ) => () =>
+	dispatch( { type: types.DUPLICATE_PAGE, payload: { pageIndex: null } } );
+
+const movePage = ( dispatch ) => ( { pageIndex, position } ) =>
+	dispatch( { type: types.MOVE_PAGE, payload: { pageIndex, position } } );
+
+const setCurrentPage = ( dispatch ) => ( { pageIndex } ) =>
+	dispatch( { type: types.SET_CURRENT_PAGE, payload: { pageIndex } } );
+
+const addElements = ( dispatch ) => ( { elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements } } );
+
+const addElement = ( dispatch ) => ( { element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex: null, elements: [ element ] } } );
+
+const addElementsToPage = ( dispatch ) => ( { pageIndex, elements } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements } } );
+
+const addElementToPage = ( dispatch ) => ( { pageIndex, element } ) =>
+	dispatch( { type: types.ADD_ELEMENTS, payload: { pageIndex, elements: [ element ] } } );
+
+const deleteElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds } } );
+
+const deleteSelectedElements = ( dispatch ) => () =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: null } } );
+
+const deleteElementById = ( dispatch ) => ( { elementId } ) =>
+	dispatch( { type: types.DELETE_ELEMENTS, payload: { elementIds: [ elementId ] } } );
+
+const duplicateElementsById = ( dispatch ) => ( { elementIds } ) =>
+	dispatch( { type: types.DUPLICATE_ELEMENTS, payload: { elementIds } } );
+
+const updateElementsById = ( dispatch ) => ( { elementIds, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds, properties } } );
+
+const updateElementById = ( dispatch ) => ( { elementId, properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: [ elementId ], properties } } );
+
+const updateSelectedElements = ( dispatch ) => ( { properties } ) =>
+	dispatch( { type: types.UPDATE_ELEMENTS, payload: { elementIds: null, properties } } );
+
+const moveElement = ( dispatch ) => ( { elementId, position } ) =>
+	dispatch( { type: types.MOVE_ELEMENT, payload: { pageIndex: null, elementId, position } } );
+
+const moveSelectedElement = ( dispatch ) => ( { position } ) =>
```

Comment:
Should this be plural?

Reply:
As stated i[n the todo for the specific underlying operation](https://github.com/ampproject/amp-wp/pull/3910/files#diff-a2eff1a69cc4279b3c8fcd9c88338b6bR28), moving/arranging elements currently only handles single element reordering.

@samitron7 is checking whether [multi-element (and multi-page) reordering is P1](https://xwp.slack.com/archives/CCCAUAH6F/p1576039887124500) (internal Slack link, sorry guests).

Reply:
I think the action should be for plural and the current UI implementation only happens to be executing it as singular. Otherwise, we don't know whether the selection is currently single or multi.

Reply:
Makes sense, I'll note it and get to it tomorrow.